app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -65,7 +65,7 @@ def x509_attributes
     {
       x509_subject: stringify_attr(x509_data.subject),
       x509_issuer: stringify_attr(x509_data.issuer),
-      x509_presented: x509_data.presented,
```

Comment:
based on the git history, it was never capable of producing a boolean.

i am a little concerned that someone coded around our bug and is now expecting the hash, but i am not sure what to do about that. we could hold off on merging this, and send out a request to the partners to check their configuration?

Reply:
another option is to keep it the hash and update the documentation, but it's a pretty weird looking hash.

```
"x509_presented": {"raw"=>true, "norm"=>nil}
```

Reply:
it looks like there is precedent for creating a config gate: https://github.com/18F/identity-idp/pull/5157 (thanks to @mitchellhenke for the reference)

Reply:
>i am a little concerned that someone coded around our bug and is now expecting the hash, but i am not sure what to do about that. we could hold off on merging this, and send out a request to the partners to check their configuration?

I could go either way, since this is definitely a bug, and I'd feel relatively comfortable saying a bug fix would be exempt from typical backwards-compatibility commitments.

But if we want to be conservative here, I think the idea of a feature flag while we collect more information about usage in the wild could make sense too.

Reply:
it looks like there's about [30 sps that request it](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:logs-insights$3FqueryDetail$3D~(end~0~start~-604800~timeType~'RELATIVE~unit~'seconds~editorString~'fields*20*40timestamp*2c*20name*2c*20coalesce*28properties.event_properties.service_provider*2c*20properties.event_properties.client_id*29*20as*20sp*2c*20properties.event_properties.scope*20as*20scope*0a*7c*20filter*20ispresent*28sp*29*20and*20properties.event_properties.success*20*3d*201*0a*7c*20filter*20properties.event_properties.scope*20like*20*2fx509*3apresented*2f*20or*20*40message*20like*20*2fx509.presented*2f*20or*20properties.event_properties.scope*20like*20*2f*20x509*2f*20or*20properties.event_properties.scope*20like*20*2fx509*20*2f*0a*7c*20stats*20count*28*2a*29*20as*20c*20by*20sp*2c*20name*2c*20scope*0a*7c*20sort*20c*20desc~queryId~'4ca3aad4a77f13cc-447e7148-425ce7d-ae24e43e-0355a7973bb34165e58075~source~(~'prod_*2fsrv*2fidp*2fshared*2flog*2fevents.log))) or the general `x509` scope, so it's quite a few possible issues

Reply:
@aduth i added the config gate and we have a ticket to follow up/remove the configuration in a month https://cm-jira.usa.gov/browse/LG-12717